greenpithumb/db_retriever.py

```diff
@@ -0,0 +1,40 @@
+class DbRetrieverBase(object):
+    """Base class for retrieving information from a database."""
+
+    def __init__(self, cursor):
+        """Creates a new database retrieval object.
+
+        Args:
+            cursor: An SQLite database cursor.
+        """
+        self._cursor = cursor
+
+
+class TemperatureRetriever(DbRetrieverBase):
+    """Retrieves timestamp and temperature readings."""
+
+    def retrieve_temperature(self, range_start, range_end):
+        """Returns timestamp and temperature info within a given range.
+
+        Args:
+            range_start: A datetime.datetime object representing the beginning
+                of the retrieval range.
+            range_end: A datetime.datetime object representing the end of the
+                retrieval range.
+
+        Returns:
+            A list of dictionaries, one per table row, of the timestamp and
+            temperature information.
+        """
+        self._cursor.execute('''SELECT * FROM temperature WHERE timestamp >= ?
+                             and timestamp <= ?''' (range_start, range_end))
+        temperature_data = self._cursor.fetchall()
+
+        temperature_data_list = []
+        for row in temperature_data:
+            d = {}
+            d['timestamp'] = row[0]
```

Comment:
Oooh namedtuples fancy. 

Agreed on explicitly converting dates to strings. I'll do that in a separate PR.